scrapers/nd/events.py

```diff
@@ -1,114 +1,174 @@
+                for item in agenda:
+                    time = datetime.datetime.strptime(item_key, "%H:%M:%S").strftime(
+                        "%I:%M %p"
                     )
+                    descr_with_time = f"[{time}]: {item['description']}"
+                    item_descr = event_obj.add_agenda_item(descr_with_time)
+                    if item["bill_name"]:
+                        item_descr.add_bill(item["bill_name"])
+                    if item["sub_com"]:
+                        item_descr.extras["sub_committee"] = item["sub_com"]
+
+            event_obj.add_source(self.url)
+
+            event_collection.append(event_obj)
+
+        return event_collection
+
+
+class BillNameScraper(HtmlPage):
+    def __init__(self, source):
+        self.source = source
+
+    def get_bill_name(self):
+        response = requests.get(self.source)
+        content = lxml.html.fromstring(response.content)
+        bill_name_tag = content.xpath(".//div[@id='content']//h3")[0]
+        bill_name = bill_name_tag.text_content().strip()
+        return bill_name
+
+
+class EventsTable(HtmlPage):
+    events_path = "/legend/committee/hearings/public-schedule/"
+    source = f"https://www.ndlegis.gov{events_path}"
+
+    def process_page(self):
+        date_span = self.root.xpath(".//h5[2]")[0].text_content()
+        start_year, end_year = re.search(r"(\d{4}).+(\d{4})", date_span).groups()
+
+        table_rows = self.root.xpath(".//tbody//tr")
+        agenda_items_list = []
+
+        for row_item in table_rows:
+            columns = row_item.xpath("./td")
+            columns_content = [x.text_content().strip() for x in columns]
+            if len(columns_content) == 7:
+                columns_content = columns_content[:-1]
+
+            bill, part_date, com, sub_com, loc, descr = columns_content
+
+            full_bill_name_match = re.search(r"[A-Z]{1,4}\s+\d+", bill)
+            if full_bill_name_match:
+                bill_name = full_bill_name_match.group()
+            else:
+                partial_bill_name_match = re.search(r"\d+", bill)
+                if partial_bill_name_match:
+                    match_in_descr = re.search(r"[A-Z]{1,4}\s{0,3}\d+", descr)
+                    if match_in_descr:
+                        bill_name = match_in_descr.group()
+                    else:
+                        bill_link = columns[0].xpath("./a")[0].get("href")
+                        bill_name_scraper = BillNameScraper(bill_link)
+                        bill_name = bill_name_scraper.get_bill_name()
+                else:
+                    bill_name = ""
+
+            date_time_parts = part_date.split()
+            month = int(re.search(r"\d+", date_time_parts[0]).group())
+            if month == 1 and not start_year == end_year:
+                event_year = end_year
+            else:
+                event_year = start_year
+            date_time_parts[0] = date_time_parts[0] + f"/{event_year}"
```

Comment:
`date_time_parts[0] += f"/{event_year}"` You can use `+=` to append to a string.

Reply:
Rookie move! [Fixed.](https://github.com/openstates/openstates-scrapers/pull/4162/commits/037209511528934a8e11f7616c83939e636b489a#diff-fa16689e28573ece469e8dfea222521c243ef203a1c8c974d48d6b050b3ed308R152)